Apps/WebClient/src/ClientApp/src/components/report/ImmunizationHistoryReportComponent.vue

```diff
@@ -141,122 +193,69 @@ export default class ImmunizationHistoryReportComponent extends Vue {
-            },
-            template: this.isDependent
-                ? TemplateType.DependentImmunization
-                : TemplateType.Immunization,
-            type: reportFormatType,
-        });
-    }
-
-    private immunizationFields: ReportField[] = [
-        {
-            key: "date",
-            thStyle: { width: "15%" },
-            thClass: this.headerClass,
-            thAttr: { "data-testid": "immunizationDateTitle" },
-            tdAttr: { "data-testid": "immunizationDateItem" },
-        },
-        {
-            key: "immunization",
-            thStyle: { width: "25%" },
-            thClass: this.headerClass,
-            thAttr: { "data-testid": "immunizationNameTitle" },
-            tdAttr: { "data-testid": "immunizationNameItem" },
-        },
-        {
-            key: "agents",
-            thStyle: { width: "45%" },
-            thClass: this.headerClass,
-            thAttr: { "data-testid": "immunizationAgentTitle" },
-            tdAttr: { "data-testid": "immunizationAgentItem" },
-        },
-        {
-            key: "provider_clinic",
-            label: "Provider / Clinic",
-            thStyle: { width: "15%" },
-            thClass: this.headerClass,
-            thAttr: { "data-testid": "immunizationProviderClinicTitle" },
-            tdAttr: { "data-testid": "immunizationProviderClinicItem" },
-        },
-    ];
-
-    private recomendationFields: ReportField[] = [
-        {
-            key: "immunization",
-            thStyle: { width: "50%" },
-            thClass: this.headerClass,
-            thAttr: { "data-testid": "recommendationTitle" },
-            tdAttr: { "data-testid": "recommendationItem" },
-        },
-        {
-            key: "due_date",
-            thStyle: { width: "50%" },
-            thClass: this.headerClass,
-            thAttr: { "data-testid": "recommendationDateTitle" },
-            tdAttr: { "data-testid": "recommendationDateItem" },
+        };
+    })
+);
+
+function retrieveImmunizations(hdid: string): Promise<void> {
+    return store.dispatch("immunization/retrieveImmunizations", { hdid });
+}
+
+function generateReport(
+    reportFormatType: ReportFormatType,
+    headerData: ReportHeader
+): Promise<RequestResult<Report>> {
+    const reportService = container.get<IReportService>(
+        SERVICE_IDENTIFIER.ReportService
+    );
+    return reportService.generateReport({
+        data: {
+            header: headerData,
+            records: immunizationItems.value,
+            recommendations: recomendationItems.value,
         },
-    ];
+        template: props.isDependent
+            ? TemplateType.DependentImmunization
+            : TemplateType.Immunization,
+        type: reportFormatType,
+    });
+}
+
+watch(isLoading, () => {
+    emit("on-is-loading-changed", isLoading.value);
+});
+
+function onIsEmptyChanged(): void {
+    emit("on-is-empty-changed", isEmpty.value && isRecommendationEmpty.value);
 }
```

Comment:
Nit: swap order to keep functions and watch() statements grouped separately.

```suggestion
function onIsEmptyChanged(): void {
    emit("on-is-empty-changed", isEmpty.value && isRecommendationEmpty.value);
}

watch(isLoading, () => {
    emit("on-is-loading-changed", isLoading.value);
});
```